### PR TITLE
Feature/episode vids

### PR DESCRIPTION
**Pull request by KMichaelMcGaw**
## Jira Ticket 
- [Title](Url)

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Intended Behavior
Adds support for parsing `podcast:alternateEnclosure` elements from the Podcast Namespace specification (https://podcastindex.org/namespace/1.0). This allows podcast episodes to offer multiple media versions with different formats (MP3, Opus, AAC, video), quality levels, and transport methods (HTTPS, IPFS, torrents, HLS streaming).

## Describe Changes Made 
- Added `alternate_enclosures` list attribute to `Item` class to store parsed alternate enclosure data
- Implemented `set_alternate_enclosure()` method to parse all alternateEnclosure attributes including:
  - Required: `type` (MIME type)
  - Optional: `length`, `bitrate`, `height`, `lang`, `title`, `rel`, `codecs`, `default`
  - Nested `podcast:source` elements with `uri` and `contentType` attributes
- Registered alternateEnclosure parser in `tag_methods` dictionary with support for multiple enclosures per item
- Updated `Item.to_dict()` to include `alternate_enclosures` in serialized output
- Maintained backward compatibility with existing `enclosure_url`, `enclosure_type`, and `enclosure_length` attributes
- Created comprehensive test RSS feed with multiple alternateEnclosure scenarios
- Added 14 unit tests covering all attributes, multiple sources, video content, and edge cases

## System Performance Impact
Minimal impact. The parser only processes alternateEnclosure elements when present in the feed. For feeds without alternateEnclosure elements, performance is unchanged. The parsing logic uses the same BeautifulSoup traversal pattern as existing elements.

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests.
- [ ] No, I am a very confident individual.

### Steps to Test
1. Run the test suite: `python -m pytest tests/test_pyPodcastParser.py::TestAlternateEnclosureFeed -v`
2. Parse a feed with alternateEnclosure elements (test feed included: `tests/test_feeds/alternate_enclosure.rss`)
3. Access `item.alternate_enclosures` to get list of alternate media versions
4. Verify each enclosure has `type`, `length`, `bitrate`, `sources`, etc.
5. Confirm standard `item.enclosure_url` still works for backward compatibility

## Quality Assurance and Verifications
Looking for screenshots, screen recordings, honorable pledges.

## Documentation Updated
- [ ] Yes
- [x] No, documentation is for the weak.

### CODE REVIEW

**Review by 187ram — APPROVED** [review]

**Review by daleyp24 — APPROVED** [review]

File: pypodcastparser/Item.py [hunk 1]
```diff
@@ -646,3 +651,81 @@ def set_interactive(self, tag):
...
+                except (ValueError, TypeError):
+                    pass
+
+            enclosure_dict["lang"] = tag.get("lang", None)
+            enclosure_dict["title"] = tag.get("title", None)
+            enclosure_dict["rel"] = tag.get("rel", None)
+            enclosure_dict["codecs"] = tag.get("codecs", None)
+
+            # Parse default attribute as boolean
+            default_value = tag.get("default", False)
```
Comment (yashtecholution) [inline pypodcastparser/Item.py]: suppose this is the case <podcast:alternateEnclosure default="false">
then

File: pypodcastparser/Item.py [hunk 2]
```diff
@@ -646,3 +651,81 @@ def set_interactive(self, tag):
...
+            if enclosure_dict["length"]:
+                try:
+                    enclosure_dict["length"] = int(enclosure_dict["length"])
+                except (ValueError, TypeError):
+                    pass
+
+            enclosure_dict["bitrate"] = tag.get("bitrate", None)
+            if enclosure_dict["bitrate"]:
+                try:
+                    enclosure_dict["bitrate"] = float(enclosure_dict["bitrate"])
```
Comment (yashtecholution) [inline pypodcastparser/Item.py:676]: can ever be values non-numeric, zero, or negative ? if yes parsing will fail

**Review by yashtecholution — COMMENTED** [review]
Should we preserve raw alternateEnclosure XML attributes without interpretation to ensure forward compatibility with future spec or vendor extensions?

Should we introduce schema versioning for alternateEnclosure data by tagging each entry with a version identifier to support safe evolution and migrations?

Should we ignore invalid media sources by skipping podcast:source entries with missing, malformed, or excessively long URLs to prevent bad data propagation?

File: pypodcastparser/Item.py [hunk 3]
```diff
@@ -646,3 +652,81 @@ def set_interactive(self, tag):
...
+            enclosure_dict["title"] = tag.get("title", None)
+            enclosure_dict["rel"] = tag.get("rel", None)
+            enclosure_dict["codecs"] = tag.get("codecs", None)
+
+            # Parse default attribute as boolean
+            default_value = tag.get("default", False)
+            if default_value:
+                enclosure_dict["default"] = True
+            else:
+                enclosure_dict["default"] = False
```
[the remark this answers is not in the document]
Comment (KMichaelMcGaw) [reply pypodcastparser/Item.py]: I like copilot better.
